### ISSUE TITLE
PV pending on GKE v1.15 with Rook 1.2.2 for Ceph

### ISSUE THREAD

**Issue by tvvignesh** [rook/rook#4778]
Hi. I am trying to use Rook with Ceph on GKE (version 1.15.4-gke.22) with Ubuntu-cos image and the PVs stay pending and does not provision the storage.

Using the default `cluster.yaml` `flex/storageclass.yaml` 

Have set flexvolume in `operator.yaml` like this:
```
name: FLEXVOLUME_DIR_PATH
          value: "/home/kubernetes/flexvolume"
```

This is my `pvc.yaml` (Tried both ReadWriteOnce and ReadWriteMany)
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc-rook-ceph-test
spec:
  storageClassName: rook-ceph-block
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
```

All resources have been applied to the cluster and no errors seen so far except for a warning in the operator which I have repored in https://github.com/rook/rook/issues/4317#issuecomment-580202578

![Screenshot from 2020-01-30 16-27-43](https://user-images.githubusercontent.com/1165845/73444751-3fe37400-437f-11ea-9271-69f66e949ec7.png)

Storage class has been applied:

![Screenshot from 2020-01-30 16-43-33](https://user-images.githubusercontent.com/1165845/73444968-b41e1780-437f-11ea-89d1-9dd4f8d49ada.png)

PVC stays in pending state for more than 30 minutes for just 1 Gi to be provisioned (ofcourse doesn't get provisioned even after that):

![Screenshot from 2020-01-30 16-28-29](https://user-images.githubusercontent.com/1165845/73444826-6d302200-437f-11ea-96d9-c0c2c55358ba.png)

Any help on how to solve this please? Thanks.

**Comment by travisn** [comment]
@tvvignesh If the PVC is staying in pending state, the most common cause is that the CSI driver directory needs to be changed for your environment. The setting can be found [here](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/operator.yaml#L216-L218). I thought this was in our docs, but I'm not seeing it now.

**Comment by tvvignesh** [comment]
@travisn Thanks. Will try it out. I had another issue which @galexrt clarified which was the fact that I had no OSDs. I used cluster-pvc yaml to make that work. Now encountering some other errors when trying to create a shared filesystems where even if you apply the filesystem.yaml file, it keeps saying filesystem myfs not found. Anyways, I guess I am doing something wrong. Will create a separate issue for that.

Thanks. Will close this issue.

**Comment by travisn** [comment]
Ok, if you don't have any OSDs that likely is also preventing the filesystem from being created.
